xrpl/transaction/main.py

```diff
@@ -52,25 +52,34 @@ def safe_sign_transaction(transaction: Transaction, wallet: Wallet) -> str:
     serialized_bytes = bytes.fromhex(serialized_for_signing)
     signature = sign(serialized_bytes, wallet.priv_key)
     transaction_json["TxnSignature"] = signature
-    return encode(transaction_json)
+    return cast(Transaction, Transaction.from_xrpl(transaction_json))
 
 
-def submit_transaction_blob(
-    transaction_blob: str,
+def submit_transaction(
+    transaction: Transaction,
     client: Client,
 ) -> Response:
     """
     Submits a transaction blob to the ledger.
 
     Args:
-        transaction_blob: the transaction blob to be submitted.
+        transaction: the transaction blob to be submitted.
```

Comment:
since we are changing the meaning of the argument to be a `Transaction` itself, let's update the docstring